lib/internal/util/comparisons.js

```diff
@@ -245,7 +246,8 @@ function objectComparisonStart(val1, val2, mode, memos) {
       if (!isPartialArrayBufferView(val1, val2)) {
         return false;
       }
-    } else if (mode === kLoose && (isFloat32Array(val1) || isFloat64Array(val1))) {
+    } else if (mode === kLoose &&
+               (isFloat16Array(val1) || isFloat32Array(val1) || isFloat64Array(val1))) {
```

Comment:
```suggestion
               (isFloat32Array(val1) || isFloat64Array(val1) || isFloat16Array(val1))) {
```

The likelihood of 32 and 64 bit are higher than 16 to being used.